app/Http/Controllers/GetPendingSubmissionsController.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers;
+
+use App\Enums\SubmissionStatuses;
+use App\Models\Submission;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+
+class GetPendingSubmissionsController
+{
+    /**
+     * Handle the incoming request.
+     *
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function __invoke(Request $request): JsonResponse
+    {
+        $submissions = Submission::where('status', SubmissionStatuses::Pending)->paginate();
+
+        return response()->json($submissions);
```

Comment:
```suggestion
        return Submission::where('status', SubmissionStatuses::Pending)->paginate();
```

Laravel ya esta preparado para convertir ya sea colecciones o modelos a una respuesta JSON, por lo que no es necesario que pases coleccion por `response()->json()`. 

Por otro lado, podrias tener un scope que ya directamente se encargue de filtrar por status 'pending'.

Reply:
En ese caso que use esa response tendria que modificar el tipo del retorno no? Es correcto igual no retornar siempre el mismo tipo de datos? Me tira un error laravel para que retorne algo de tipo LengthAwarePaginator

Reply:
Si, `paginate()` te retorna una instancia de `LengthAwarePaginator`, que es lo que Laravel te esta diciendo que cambias para el tipo de retorno.
A que te referis con "Es correcto igual no retornar siempre el mismo tipo de datos?"? No entiendo la pregunta